lib/cucumber/multiline_argument/data_table.rb

```diff
@@ -692,7 +692,18 @@ def index
         end
 
         def width
-          map{|cell| cell.value ? escape_cell(cell.value.to_s).unpack('U*').length : 0}.max
+          map {|cell|
+            if cell.value
+              escaped = escape_cell(cell.value.to_s)
+              if defined?(Unicode)
+                Unicode.width(escaped, Cucumber.treats_ambiguous_as_fullwidth)
```

Comment:
Instead of using the global option here, let's accept this specific option as a constructor parameter (using an options hash or keyword argument would be best IMO) into the `DataTable` and pass it down to the `Cell`.

Wherever we construct instances of `DataTable` we'll almost certainly have access to an instance of the `configuration` where we can read that setting from.